packages/mask/src/extension/popups/pages/Wallet/Transfer/FungibleTokenSection.tsx

```diff
@@ -192,18 +194,19 @@ export const FungibleTokenSection = memo(function FungibleTokenSection() {
                             className={classes.maxButton}
                             onClick={() => {
                                 if (!balance || !token?.decimals) return
-                                setAmount(leftShift(balance, token.decimals).toFixed())
+                                setAmount(uiTokenBalance)
                             }}>
                             {t('max')}
                         </Typography>
                     }
                     value={amount}
                     onChange={(e) => {
-                        if (!balance || !token?.decimals || !e.target.value) {
-                            setAmount(e.target.value)
+                        let value = e.target.value
+                        if (!balance || !token?.decimals || !value) {
+                            setAmount(value)
                             return
                         }
-                        const value = BigNumber.min(e.target.value, leftShift(balance, token.decimals)).toFixed()
+                        value = isLessThan(value, uiTokenBalance) ? value : uiTokenBalance
```

Comment:
```ts
const value = e.target.value
if (!balance || !token?.decimals || !value) {
    setAmount(value)
    return
}
setAmount(max(value, uiTokenBalance))
```

Reply:
You mean `setAmount(min(value, uiTokenBalance))`, right?

If user intent to input `0.1`, during user inputing number, value could be `0.`, 
`min('0.', 1))` will be `0`, then user will never have chance to input `0.1`

Reply:
Oops, another try?
```ts
return BigNumber.min(e.target.value, leftShift(balance, token.decimals), uiTokenBalance).toFixed()
```